pkg/clusteragent/admission/mutate/config/config.go

```diff
@@ -65,29 +65,37 @@ const (
 
 // Webhook is the webhook that injects DD_AGENT_HOST and DD_ENTITY_ID into a pod
 type Webhook struct {
-	name            string
-	isEnabled       bool
-	endpoint        string
-	resources       []string
-	operations      []admissionregistrationv1.OperationType
-	mode            string
-	wmeta           workloadmeta.Component
-	injectionFilter mutatecommon.InjectionFilter
-	datadogConfig   config.Component
+	name              string
+	isEnabled         bool
+	endpoint          string
+	resources         []string
+	operations        []admissionregistrationv1.OperationType
+	mode              string
+	wmeta             workloadmeta.Component
+	injectionFilter   mutatecommon.InjectionFilter
+	localServiceName  string
+	traceAgentSocket  string
+	dogStatsDSocket   string
+	socketPath        string
+	typeSocketVolumes bool
 }
 
 // NewWebhook returns a new Webhook
 func NewWebhook(wmeta workloadmeta.Component, injectionFilter mutatecommon.InjectionFilter, datadogConfig config.Component) *Webhook {
 	return &Webhook{
-		name:            webhookName,
-		isEnabled:       datadogConfig.GetBool("admission_controller.inject_config.enabled"),
-		endpoint:        datadogConfig.GetString("admission_controller.inject_config.endpoint"),
-		resources:       []string{"pods"},
-		operations:      []admissionregistrationv1.OperationType{admissionregistrationv1.Create},
-		mode:            datadogConfig.GetString("admission_controller.inject_config.mode"),
-		wmeta:           wmeta,
-		injectionFilter: injectionFilter,
-		datadogConfig:   datadogConfig,
+		name:              webhookName,
+		isEnabled:         datadogConfig.GetBool("admission_controller.inject_config.enabled"),
+		endpoint:          datadogConfig.GetString("admission_controller.inject_config.endpoint"),
+		resources:         []string{"pods"},
+		operations:        []admissionregistrationv1.OperationType{admissionregistrationv1.Create},
+		mode:              datadogConfig.GetString("admission_controller.inject_config.mode"),
+		wmeta:             wmeta,
+		injectionFilter:   injectionFilter,
+		localServiceName:  datadogConfig.GetString("admission_controller.inject_config.local_service_name"),
+		traceAgentSocket:  datadogConfig.GetString("admission_controller.inject_config.trace_agent_socket"),
+		dogStatsDSocket:   datadogConfig.GetString("admission_controller.inject_config.dogstatsd_socket"),
+		socketPath:        datadogConfig.GetString("admission_controller.inject_config.socket_path"),
+		typeSocketVolumes: datadogConfig.GetBool("admission_controller.inject_config.type_socket_volumes"),
```

Comment:
regrouping the fields coming from the `datadogConfig` in a sub structure can help for the readability